### PR TITLE
finished task

### PR DESCRIPTION
**Pull request by Dima23-ops5**

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,7 +1,7 @@
-spring.application.name=StringBootIntro
-spring.datasource.url=jdbc:mysql://localhost:3306/book
-spring.datasource.username=root
-spring.datasource.password=bookPass
-spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
-spring.jpa.hibernate.ddl-auto=create-drop
-spring.jpa.show-sql=true
\ No newline at end of file
+spring.application.name = StringBootIntro
```
Comment (Elena-Bruyako) [inline src/main/resources/application.properties]: Revert these changes

File: src/main/java/mate/academy/StringBootIntro/model/Book.java [hunk 2]
```diff
@@ -83,7 +83,7 @@ public void setCoverImage(String coverImage) {
 
     @Override
     public String toString() {
-        return "Book = {"
+        return "Book = { "
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/model/Book.java]: Let's use Lombok annotations for generating get and set methods and toString also

File: src/main/java/mate/academy/StringBootIntro/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -31,7 +31,7 @@ public Book save(Book book) {
             if (transaction != null) {
                 transaction.rollback();
             }
-            throw new RuntimeException("Cannot save book", e);
+            throw new RuntimeException("Can`t save book", e);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/repository/impl/BookRepositoryImpl.java]: Use `@RequiredArgsConstructor` and remove constructor

Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/repository/impl/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can`t save book", e);
```

File: src/main/java/mate/academy/StringBootIntro/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -9,6 +9,7 @@
 
 @Service
 public class BookServiceImpl implements BookService {
+
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/service/impl/BookServiceImpl.java]: The same as above about constructor
Remove blank line

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
need to fix comments

File: src/main/java/mate/academy/StringBootIntro/model/Book.java [hunk 5]
```diff
@@ -25,72 +30,16 @@ public class Book {
...
-                + "id = " + id
-                + ", title = " + title
-                + ", author = " + author
-                + ", isbn = " + isbn
-                + ", price = " + price
-                + ", description = " + description
-                + ", coverImage = " + coverImage
-                + "}";
-    }
+    //@Override
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/model/Book.java]: remove commented code

File: src/main/java/mate/academy/StringBootIntro/repository/BookRepository.java [hunk 6]
```diff
@@ -5,5 +5,6 @@
 
 public interface BookRepository {
     Book save(Book book);
+
     List findAll();
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/repository/BookRepository.java]:
```suggestion
    List<Book> findAll();
```

File: src/main/java/mate/academy/StringBootIntro/service/BookService.java [hunk 7]
```diff
@@ -6,5 +6,6 @@
 
 public interface BookService {
     Book save(Book book);
+
     List findAll();
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/StringBootIntro/service/BookService.java]:
```suggestion
    List<Book> findAll();
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
need to fix

File: src/main/java/mate/academy/StringBootIntro/model/Book.java [hunk 8]
```diff
@@ -6,10 +6,12 @@
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.Table;
 import jakarta.persistence.Id;
-import java.math.BigDecimal;
+import lombok.Data;
 
+import java.math.BigDecimal;
 @Entity
 @Table(name = "Books")
+@Data
```
Comment (okuzan) [inline src/main/java/mate/academy/StringBootIntro/model/Book.java]: Don't use @Data on JPA entities, it can cause troubles, use @ Getter @ Setter separately

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/StringBootIntro/model/Book.java (hunk 8 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/StringBootIntro/model/Book.java]: Not fixed

File: src/main/java/mate/academy/StringBootIntro/service/BookService.java (hunk 7 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/StringBootIntro/service/BookService.java]: not fixed

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
fix comments

**Review by Elena-Bruyako — APPROVED** [review]
Good job!
